contentcuration/contentcuration/static/js/edit_channel/views.js

```diff
@@ -930,31 +936,33 @@ var BaseWorkspaceListView = BaseEditableListView.extend({
 	add_exercise:function(){
 		var UploaderViews = require("edit_channel/uploader/views");
 		var self = this;
-		this.collection.create_new_node({
-            "kind":"exercise",
-            "title": (this.model.get('parent'))? this.model.get('title') + " " + this.get_translation("exercise_title") : this.get_translation("exercise_title"), // Avoid having exercises prefilled with 'email clipboard'
-            "author": State.preferences.author || "",
-            "aggregator": State.preferences.aggregator || "",
-            "provider": State.preferences.provider || "",
-            "copyright_holder": State.preferences.copyright_holder || "",
-            "license_name": State.preferences.license,
-            "license_description": State.preferences.license_description || ""
-        }).then(function(new_exercise){
-        	var edit_collection = new Models.ContentNodeCollection([new_exercise]);
-	        $("#main-content-area").append("<div id='dialog'></div>");
+		State.Store.dispatch('usePrimaryModal', () => {
+			var metadata_modal = new UploaderViews.MetadataModalView({
+				collection: new Models.ContentNodeCollection([]),
+				model: self.model,
+				new_content: true,
+				new_exercise: true,
+				onsave: self.reload_ancestors,
+				onnew:self.add_nodes,
+				allow_edit: true,
+				isclipboard: self.isclipboard
+			});
+			this.collection.create_new_node({
+				"kind":"exercise",
```

Comment:
To quote keys or not to quote keys? That is the question.
Whether 'tis nobler in the mind to suffer
The slings and arrows of inconsistent quotation marks,
Or, by opposing, end them?